R/volcanoPlot.R

```diff
@@ -114,244 +109,373 @@ volcanoPlot <- function(contrastDF,
-                                                x     = foldChangeLines)
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color = symbolFill[which(groupNames == "Decreased")],
-                                                width = refLineThickness,
-                                                x     = -1*foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[2],
+                                                 width       = refLineThickness,
+                                                 x           = foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[1],
+                                                 width       = refLineThickness,
+                                                 x           = -foldChangeLines)
+        }
+
+        events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
+                                                if (o != null && o != false) {
+                                                  if (o.y != null &&
+                                                      o.y.data != null &&
+                                                      o.y.smps != null) {
+                                                      info = '<b>' + o.y.vars[0]  + '</b>' + '<br/>' +
+                                                             '<i>' + o.z.Group  + '</i><br/>' +
+                                                             '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
+                                                             '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1] ;
+                                                      if (o.z != null && o.z['GeneName'] != null) {
+                                                        info  = info + '<br/>' +
+                                                              '<b> Symbol</b>' + ': ' + o.z['GeneName'] ;
+                                                      }
+                                                    t.showInfoSpan(e, info);
+                                                  }
+                                                }; }}")
+
+        cx.data <- contrastDF %>% dplyr::select(all_of(logRatioCol), negLog10P)
+
+        if (sizeByIntensity) {
+            var.annot <- contrastDF %>% dplyr::select(Group,LogInt)
+            sizeBy <- "LogInt"
+            showSizeLegend <- TRUE
+        } else {
+            var.annot <- contrastDF %>%
+                dplyr::select(Group)
+            sizeBy <- "Group"
+            showSizeLegend <- FALSE
         }
 
-        # Assign null if footnote is missing
-        if (missing(footnote)) {
-            footnote <- NULL
+        if (!missing(geneSymCol)) {
```

Comment:
can you do a merge with rownames merge (a,b, by =0). by =0 indicates merging based on row names